src/orion/executor/ray_backend.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import traceback
+
+from orion.executor.base import (
+    AsyncException,
+    AsyncResult,
+    BaseExecutor,
+    ExecutorClosed,
+    Future,
+)
+
+try:
+    import ray
+
+    HAS_RAY = True
+except ImportError:
+    HAS_RAY = False
+
+logger = logging.getLogger(__name__)
+
+
+class _Future(Future):
+    def __init__(self, future):
+        self.future = future
+        self.exception = None
+
+    def get(self, timeout=None):
+        if self.exception:
+            raise self.exception
+        try:
+            return ray.get(self.future, timeout=timeout)
+        except ray.exceptions.GetTimeoutError as e:
+            raise TimeoutError() from e
+
+    def wait(self, timeout=None):
+        try:
+            ray.get(self.future, timeout=timeout)
+        except ray.exceptions.GetTimeoutError:
+            pass
+        except Exception as e:
+            self.exception = e
+
+    def ready(self):
+        obj_ready = ray.wait([self.future])
```

Comment:
Would this work if we call ready twice? Can we call ray.wait twice on the same object? Also, `wait` will block until the future is done. `ready` should not be a blocking call, hence we should give a timeout arbitrarily small here (ex: 0.01 seconds).

Reply:
So, here I did some tests and it can be called twice on the same object. The wait function does not block until the future is done, it simply returns the list of futures that are ready.

Reply:
Yes it returns them again. Just tested it.